collector/infiniband_linux.go

```diff
@@ -144,6 +145,11 @@ func infinibandPorts(infinibandPath, device string) ([]string, error) {
 func readMetric(directory, metricFile string) (uint64, error) {
 	metric, err := readUintFromFile(filepath.Join(directory, metricFile))
```

Comment:
I think we should rather read the file first to check if it contains `N/A (no PMA)` instead of having this error out and check the error message.

Reply:
@discordianfish **readUintFromFile** is defined at `collector/helper.go`, and used by many other collectors. IMHO, I think this is the place for handling it, as it only happens when `i40iw0` devices are present, and not everywhere. Thoughts?